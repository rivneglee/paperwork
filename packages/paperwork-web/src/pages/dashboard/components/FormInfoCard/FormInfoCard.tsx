import React from 'react';
import { Button, Card } from '@paperwork/ui-widgets';

import './FormInfoCard.scss';
const img = require('../../../../assets/compute-data.svg');

interface Props {
  onCreateNewForm: () => void;
}

export default ({ onCreateNewForm }: Props) => (
    <Card>
      <div className="pwapp-form-info">
        <div className="pwapp-form-info__primary">
          <div className="pwapp-form-info__title">Collecting Data via Form</div>
          <div className="pwapp-form-info__description">
            You can customise forms on demand and send them to people to collect data you are interested.
          </div>
          <div>
            <Button onClick={onCreateNewForm} size="xl" color="primary">Create a form</Button>
          </div>
        </div>
        <div className="pwapp-form-info__secondary">
          <img src={img}/>
        </div>
      </div>
    </Card>
);
