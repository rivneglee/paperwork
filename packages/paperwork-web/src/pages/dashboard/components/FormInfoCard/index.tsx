export { default as FormInfoCard } from './FormInfoCard';
