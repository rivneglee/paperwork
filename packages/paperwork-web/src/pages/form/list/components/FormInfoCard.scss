.pwapp-form-card {
  position: relative;
  min-width: 200px;
  min-height: 190px;
  height: auto;
  width: 100%;
  box-shadow: 0 1px 2px 0 rgba(46,62,79,0.1);

  @media screen and (min-width: 520px) {
    min-height: 160px;
  }

  &__part {
    width: 100%;
    height: 100%;

    &__side {
      position: absolute;
      left: 0;
      top: 0;
      width: 100%;
      height: 100%;
      background: transparent;
    }

    &__inner {
      overflow: hidden;
      position: relative;
      height: 100%;
      cursor: pointer;
      transition: box-shadow 0.5s;
      &:hover {
        box-shadow: 0 26px 56px 0 #0000005e
      }
    }
  }

  &__face {
    box-sizing: border-box;
    padding: 20px 20px 15px;

    &__colored-side {
      position: absolute;
      left: 0;
      top: 0;
      width: 7px;
      height: 100%;
      transition: width 0.3s;
      background: #9C27B0;

      &--pink {
        background: #E91E63;
      }
      &--red {
        background: #F44336;
      }
      &--purple {
        background: #9C27B0;
      }
      &--blue {
        background: #2196F3;
      }
      &--blue-grey {
        background: #607D8B;
      }
      &--indigo {
        background: #3F51B5;
      }
      &--cyan {
        background: #00BCD4;
      }
      &--teal {
        background: #009688;
      }
      &--green {
        background: #4CAF50;
      }
      &--light-green {
        background: #8BC34A;
      }
      &--lime {
        background: #CDDC39;
      }
      &--yellow {
        background: #FFEB3B;
      }
      &--amber {
        background: #FFC107;
      }
      &--orange {
        background: #FF9800;
      }
      &--deep-orange {
        background: #FF5722;
      }
      &--brown {
        background: #795548;
      }
    }

    &__status {
      color: #555A5F;
      display: inline-block;
    }

    &__heading {
      position: relative;
      left: 10px;
      width: 90%;
      color: #555A5F;
      font-size: 13px;
      overflow: hidden;

      &:before {
        content: "";
        position: absolute;
        left: 0;
        top: 50%;
        width: 100%;
        height: 1px;
        background: #ECECEC;
      }
    }

    &__title {
      margin: 0 0 15px;
      font-size: 16px;
      font-weight: bold;
      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;
    }

    &__stats {
      position: relative;
      width: 100%;
      display: flex;
      flex-wrap: wrap;
      top: 25px;
      left: 10px;
      color: #B5B5B5;
      text-transform: uppercase;
      font-size: 12px;
    }

    &__data {
      margin-right: 1rem;
      p {
        margin: .1rem 1.2rem 0rem;
        font-size: 15px;
        color: #ADADAD;
        font-weight: bold;
      }
    }

    &__data-heading {
      display: flex;
      align-items: center;
    }

     &__icon {
       fill: #ADADAD;
       margin-right: .2rem;
     }
  }
}
