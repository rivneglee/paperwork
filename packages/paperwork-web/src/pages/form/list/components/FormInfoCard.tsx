import React, { FunctionComponent } from 'react';
import classNames from 'classnames';
import { Badge, IconButton, Icons, Tooltip } from '@paperwork/ui-widgets';

import './FormInfoCard.scss';
import ProgressChart from './ProgressChart';
import { Status } from '../../../../schema/Form';
import QRCode from '../../../../components/QRCode/QRCode';

export interface FormItem {
  id: string;
  name: string;
  status: string;
  isPublic: boolean;
  createdAt: string;
  closeDate: string;
  targetCommits: string | number;
  maxCommits: string | number;
  receivedCommits: number;
  targetGap: number;
  maxGap: number;
  theme?: string;
  progress?: string;
  statusBadgeColor?: any;
  newCommitUrl: string;
}

interface Props extends FormItem {
  onEdit: (id: string) => void;
  onViewCommits: (formId: string) => void;
}

const FormInfoCard: FunctionComponent<Props> = ({
  id,
  name,
  status,
  isPublic,
  createdAt,
  closeDate,
  targetCommits,
  maxCommits,
  receivedCommits,
  statusBadgeColor,
  theme,
  targetGap,
  maxGap,
  progress,
  newCommitUrl,
  onEdit,
  onViewCommits,
}) => {
  const showProgress = targetGap > 0 || maxGap > 0;
  const isClosed = status.toLowerCase() !== Status.CLOSED;
  const onClickLink = () => window.open(newCommitUrl, '_blank');
  return (
    <section className="pwapp-form-card">
      <section className="pwapp-form-card__part">
        <div className="pwapp-form-card__part__side">
          <div className="pwapp-form-card__part__inner pwapp-form-card__face">
            <div className={
              classNames(
                'pwapp-form-card__face__colored-side',
                theme && `pwapp-form-card__face__colored-side--${theme}`,
              )}
            ></div>
            <div onClick={() => onEdit(id)}>
              <div className="pwapp-form-card__face__heading">
                <p className="pwapp-form-card__face__title">{name}</p>
                {createdAt}
                <Badge color={statusBadgeColor}>{status}</Badge>
                {
                  isPublic && <Badge color="primary">PUBLIC</Badge>
                }
              </div>
              <div className="pwapp-form-card__face__stats">
                <div className="pwapp-form-card__face__data">
                  <div className="pwapp-form-card__face__data-heading">
                    <Icons.Submit className="pwapp-form-card__face__icon"/>
                    Commits
                  </div>
                  <p>{receivedCommits}</p>
                </div>
                <div className="pwapp-form-card__face__data">
                  <div className="pwapp-form-card__face__data-heading">
                    <Icons.Target className="pwapp-form-card__face__icon"/>
                    Target
                  </div>
                  <p>{targetCommits}</p>
                </div>
                <div className="pwapp-form-card__face__data">
                  <div className="pwapp-form-card__face__data-heading">
                    <Icons.Dashboard className="pwapp-form-card__face__icon"/>
                    MAX
                  </div>
                  <p>{maxCommits}</p>
                </div>
                <div className="pwapp-form-card__face__data">
                  <div className="pwapp-form-card__face__data-heading">
                    <Icons.Timer className="pwapp-form-card__face__icon"/>
                    Close date
                  </div>
                  <p>{closeDate}</p>
                </div>
              </div>
            </div>
            <div className="pwapp-form-card__buttons">
              <Tooltip placement="top" content="Open form viewer">
                <IconButton onClick={onClickLink}><Icons.Link/></IconButton>
              </Tooltip>
              <Tooltip placement="top" content={
                <QRCode url={newCommitUrl}/>
              }>
                <IconButton><Icons.QRCode/></IconButton>
              </Tooltip>
              <Tooltip placement="top" content="Commits">
                <IconButton onClick={() => onViewCommits(id)}><Icons.Commit/></IconButton>
              </Tooltip>
              {
                isClosed && (
                  <Tooltip placement="top" content="Close from">
                    <IconButton><Icons.Archive/></IconButton>
                  </Tooltip>
                )
              }
            </div>
          </div>
        </div>
      </section>
      {
        showProgress && (
          <ProgressChart targetGap={targetGap} progress={progress} maxGap={maxGap} receivedCommits={receivedCommits}/>
        )
      }
    </section>
  );
};

export default FormInfoCard;
