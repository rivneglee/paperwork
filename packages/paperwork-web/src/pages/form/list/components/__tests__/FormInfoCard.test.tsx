import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import FormInfoCard from '../FormInfoCard';

describe('FormInfoCard', () => {
  const onEdit = jest.fn();
  const onViewCommits = jest.fn();
  it('should render public badge', () => {
    const wrapper: ShallowWrapper = shallow(
      <FormInfoCard
        id="1"
        name="Form"
        status="open"
        isPublic={true}
        createdAt="30/06/2020"
        closeDate="--"
        targetCommits="--"
        maxCommits="--"
        receivedCommits={10}
        targetGap={10}
        maxGap={100}
        onEdit={onEdit}
        onViewCommits={onViewCommits}
        newCommitUrl="/foo"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should not render public badge', () => {
    const wrapper: ShallowWrapper = shallow(
      <FormInfoCard
        id="1"
        name="Form"
        status="open"
        isPublic={false}
        createdAt="30/06/2020"
        closeDate="--"
        targetCommits="--"
        maxCommits="--"
        receivedCommits={10}
        targetGap={10}
        maxGap={100}
        onEdit={onEdit}
        onViewCommits={onViewCommits}
        newCommitUrl="/foo"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should not render progress chart', () => {
    const wrapper: ShallowWrapper = shallow(
      <FormInfoCard
        id="1"
        name="Form"
        status="open"
        isPublic={false}
        createdAt="30/06/2020"
        closeDate="--"
        targetCommits="--"
        maxCommits="--"
        receivedCommits={10}
        targetGap={0}
        maxGap={0}
        onEdit={onEdit}
        onViewCommits={onViewCommits}
        newCommitUrl="/foo"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
