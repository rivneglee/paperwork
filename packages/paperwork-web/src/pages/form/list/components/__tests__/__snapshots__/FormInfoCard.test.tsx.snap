// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInfoCard should not render progress chart 1`] = `
<section
  className="pwapp-form-card"
>
  <section
    className="pwapp-form-card__part"
  >
    <div
      className="pwapp-form-card__part__side"
    >
      <div
        className="pwapp-form-card__part__inner pwapp-form-card__face"
      >
        <div
          className="pwapp-form-card__face__colored-side"
        />
        <div
          onClick={[Function]}
        >
          <div
            className="pwapp-form-card__face__heading"
          >
            <p
              className="pwapp-form-card__face__title"
            >
              Form
            </p>
            30/06/2020
            <F>
              open
            </F>
          </div>
          <div
            className="pwapp-form-card__face__stats"
          >
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Submit
                  className="pwapp-form-card__face__icon"
                />
                Commits
              </div>
              <p>
                10
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Target
                  className="pwapp-form-card__face__icon"
                />
                Target
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Dashboard
                  className="pwapp-form-card__face__icon"
                />
                MAX
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Timer
                  className="pwapp-form-card__face__icon"
                />
                Close date
              </div>
              <p>
                --
              </p>
            </div>
          </div>
        </div>
        <div
          className="pwapp-form-card__buttons"
        >
          <Component
            content="Open form viewer"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Link />
            </p>
          </Component>
          <Component
            content={
              <Unknown
                url="/foo"
              />
            }
            placement="top"
          >
            <p>
              <QRCode />
            </p>
          </Component>
          <Component
            content="Commits"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Commit />
            </p>
          </Component>
          <Component
            content="Close from"
            placement="top"
          >
            <p>
              <Archive />
            </p>
          </Component>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`FormInfoCard should not render public badge 1`] = `
<section
  className="pwapp-form-card"
>
  <section
    className="pwapp-form-card__part"
  >
    <div
      className="pwapp-form-card__part__side"
    >
      <div
        className="pwapp-form-card__part__inner pwapp-form-card__face"
      >
        <div
          className="pwapp-form-card__face__colored-side"
        />
        <div
          onClick={[Function]}
        >
          <div
            className="pwapp-form-card__face__heading"
          >
            <p
              className="pwapp-form-card__face__title"
            >
              Form
            </p>
            30/06/2020
            <F>
              open
            </F>
          </div>
          <div
            className="pwapp-form-card__face__stats"
          >
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Submit
                  className="pwapp-form-card__face__icon"
                />
                Commits
              </div>
              <p>
                10
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Target
                  className="pwapp-form-card__face__icon"
                />
                Target
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Dashboard
                  className="pwapp-form-card__face__icon"
                />
                MAX
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Timer
                  className="pwapp-form-card__face__icon"
                />
                Close date
              </div>
              <p>
                --
              </p>
            </div>
          </div>
        </div>
        <div
          className="pwapp-form-card__buttons"
        >
          <Component
            content="Open form viewer"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Link />
            </p>
          </Component>
          <Component
            content={
              <Unknown
                url="/foo"
              />
            }
            placement="top"
          >
            <p>
              <QRCode />
            </p>
          </Component>
          <Component
            content="Commits"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Commit />
            </p>
          </Component>
          <Component
            content="Close from"
            placement="top"
          >
            <p>
              <Archive />
            </p>
          </Component>
        </div>
      </div>
    </div>
  </section>
  <ProgressChart
    maxGap={100}
    receivedCommits={10}
    targetGap={10}
  />
</section>
`;

exports[`FormInfoCard should render public badge 1`] = `
<section
  className="pwapp-form-card"
>
  <section
    className="pwapp-form-card__part"
  >
    <div
      className="pwapp-form-card__part__side"
    >
      <div
        className="pwapp-form-card__part__inner pwapp-form-card__face"
      >
        <div
          className="pwapp-form-card__face__colored-side"
        />
        <div
          onClick={[Function]}
        >
          <div
            className="pwapp-form-card__face__heading"
          >
            <p
              className="pwapp-form-card__face__title"
            >
              Form
            </p>
            30/06/2020
            <F>
              open
            </F>
            <F
              color="primary"
            >
              PUBLIC
            </F>
          </div>
          <div
            className="pwapp-form-card__face__stats"
          >
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Submit
                  className="pwapp-form-card__face__icon"
                />
                Commits
              </div>
              <p>
                10
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Target
                  className="pwapp-form-card__face__icon"
                />
                Target
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Dashboard
                  className="pwapp-form-card__face__icon"
                />
                MAX
              </div>
              <p>
                --
              </p>
            </div>
            <div
              className="pwapp-form-card__face__data"
            >
              <div
                className="pwapp-form-card__face__data-heading"
              >
                <Timer
                  className="pwapp-form-card__face__icon"
                />
                Close date
              </div>
              <p>
                --
              </p>
            </div>
          </div>
        </div>
        <div
          className="pwapp-form-card__buttons"
        >
          <Component
            content="Open form viewer"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Link />
            </p>
          </Component>
          <Component
            content={
              <Unknown
                url="/foo"
              />
            }
            placement="top"
          >
            <p>
              <QRCode />
            </p>
          </Component>
          <Component
            content="Commits"
            placement="top"
          >
            <p
              onClick={[Function]}
            >
              <Commit />
            </p>
          </Component>
          <Component
            content="Close from"
            placement="top"
          >
            <p>
              <Archive />
            </p>
          </Component>
        </div>
      </div>
    </div>
  </section>
  <ProgressChart
    maxGap={100}
    receivedCommits={10}
    targetGap={10}
  />
</section>
`;
